packages/expo-splash-screen/ios/EXSplashScreen/EXSplashScreenService.h

```diff
@@ -55,7 +55,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @end
 
-@interface EXSplashScreenService : UMSingletonModule <EXSplashScreenService, UIApplicationDelegate>
```

Comment:
quick question -- if we make the below change in `EXSplashScreenViewNativeProvider.m`, do we still need to remove this auto-initialization?

Reply:
It depends on how `expo-updates` handle `RootViewController`? 🤔
Auto-initialization expects the application to have `RootViewController` assigned to the main `Window` beforehand and mostly that's it when we talk about `expo-splash-screen`.
However as I'm reading `expo-updates` I see that it registers it's own splash screen view that is independent to `expo-splash-screen`'s one.
Shouldn't we try to unify this flow?
I mean, I believe `startAndShowLaunchScreen` might be changed to plain `start` as `expo-splash-screen` can handle what's happening in this method.

I might be missing sth here, but with current change proposal we have the following scenario:
1. App launches
2. `expo-updates` creates it's own ViewController with SplashScreen View and assigns it to the main UIWindow
3. is `expo-updates` changing the main ViewController once the update is being downloaded and loaded?
4. `expo-updates` finishes it's work and calls `expo-splash-screen` to do it work (register SplashScreen on UIWindow's main ViewController)

I imagine we want sth like this:
1. App launches
2. `expo-updates` starts it's work and simultaneously `expo-splash-screen` registers it's view on main ViewController
3. is `expo-updates` switching UIWindow's ViewController during it's runtime (eg. when new version of app is loaded)?
	- if it is  ten it should call `expo-splash-screen` to register it's view on this new main ViewController
	- if it's not nothing should be done
4. as `expo-splash-screen` listens for bridge events it will autohide once the js app is rendered.